File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/model/ModerationResult.java
package com.sdkwork.ai.model;

public class ModerationResult {
    private Boolean flagged;
    private Categories categories;
    private CategoryScores categoryScores;

    public Boolean getFlagged() {
        return this.flagged;
    }
    
    public void setFlagged(Boolean flagged) {
        this.flagged = flagged;
    }

    public Categories getCategories() {
        return this.categories;
    }
    
    public void setCategories(Categories categories) {
        this.categories = categories;
    }

    public CategoryScores getCategoryScores() {
        return this.categoryScores;
    }
    
    public void setCategoryScores(CategoryScores categoryScores) {
        this.categoryScores = categoryScores;
    }
}

File: sdkwork-ai-sdk-typescript/src/types/moderation-result.ts
import type { Categories } from './categories';
import type { CategoryScores } from './category-scores';

/** Moderation result for a single input */
export interface ModerationResult {
  /** Whether the content violates OpenAI's usage policies */
  flagged?: boolean;
  /** A list of the categories, and whether they are flagged or not */
  categories?: Categories;
  /** A list of the categories along with their scores as predicted by model */
  category_scores?: CategoryScores;
}
